### PR TITLE
Fixed issue [security] #19169: Privilege escalation to edit survey

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: Permission check must be done before updating survey

### CODE REVIEW

File: application/controllers/SurveyAdministrationController.php [hunk 1]
```diff
@@ -813,18 +813,18 @@ public function actionChangeMultipleSurveyGroup()
 
         foreach ($aSIDs as $iSurveyID) {
             $oSurvey = Survey::model()->findByPk((int)$iSurveyID);
-            $oSurvey->gsid = $iSurveyGroupId;
```
Comment (gabrieljenik) [inline application/controllers/SurveyAdministrationController.php:816]: Why moving this line? Intended?

Comment (Shnoulle) [reply application/controllers/SurveyAdministrationController.php:816]: It's **the** issue if you set the gid before testing is user have Permission : it was tested with new gid. If user have Survey permission onnew gid : return true.

It must be tested with current gid

Comment (Shnoulle) [reply application/controllers/SurveyAdministrationController.php:816]: See : https://github.com/LimeSurvey/LimeSurvey/blob/d3fb278e57e5561a62b1e05c619db5b1c5fbef73/application/models/Survey.php#L2279

return with new gid here : https://github.com/LimeSurvey/LimeSurvey/blob/d3fb278e57e5561a62b1e05c619db5b1c5fbef73/application/models/Survey.php#L2294

Comment (Shnoulle) [reply application/controllers/SurveyAdministrationController.php:816]: https://github.com/LimeSurvey/LimeSurvey/pull/3547/commits/173c2d0755b278cace63511ce9e9bad03810a270

Comment (gabrieljenik) [reply application/controllers/SurveyAdministrationController.php:816]: > Why after ? We already check if user have permission to view surveygroup in Survey model.

Ok, didn't recall it.

**Comment by Shnoulle** [comment]
Code review done ?

**Pull request merged**